simpleflow/task.py

```diff
@@ -68,6 +69,11 @@ def __repr__(self):
 
     def execute(self):
         method = self.activity.callable
+        # NB: the following line attaches some *state* to the callable wrapped
+        # in this instance, so it can be used directly for advanced usage. This
+        # works well because we don't do multithreading, but if we ever do,
+        # DANGER!
+        method.context = self.context
         if hasattr(method, 'execute'):
             return method(*self.args, **self.kwargs).execute()
```

Comment:
We can be safer on activity task classes by setting the context only on the instance ?

```python
         if hasattr(method, 'execute'):
              task = method(*self.args, **self.kwargs)
              task.context = self.context
              task.execute()
         else:
               method.context = self.context
               return method(*self.args, **self.kwargs).execute()